src/frontend/js/components/ReferralDetail/CloseReferralModal.tsx

```diff
@@ -0,0 +1,198 @@
+    description:
+      'Label for the change explanation field in the modal to close the referral.',
+    id: 'components.ReferralDetail.CloseReferralModal.formLabelExplanation',
+  },
+  modalTitle: {
+    defaultMessage: 'Close the referral.',
+    description:
+      'Title for the modal that allows unit members to close the referral.',
+    id: 'components.ReferralDetail.CloseReferralModal.modalTitle',
+  },
+  update: {
+    defaultMessage: 'Close referral',
+    description: 'Button toclose the referral in the relevant modal.',
+    id: 'components.ReferralDetail.CloseReferralModal.update',
+  },
+});
+
+// The `setAppElement` needs to happen in proper code but breaks our testing environment.
+// This workaround is not satisfactory but it allows us to both test <SendAnswerModal />
+// and avoid compromising accessibility in real-world use.
+const isTestEnv = typeof jest !== 'undefined';
+if (!isTestEnv) {
+  ReactModal.setAppElement('#app-root');
+}
+
+interface CloseReferralModalProps {
+  isCloseReferralModalOpen: boolean;
+  setIsCloseReferralModalOpen: (isOpen: boolean) => void;
+  referral: Referral;
+}
+
+export const CloseReferralModal: React.FC<CloseReferralModalProps> = ({
+  isCloseReferralModalOpen,
+  setIsCloseReferralModalOpen,
+  referral,
+}) => {
+  const seed = useUIDSeed();
+  const mutation = useReferralAction();
+
+  // Keep track of the first form submission to show validation errors
+  const [isFormCleaned, setIsFormCleaned] = useState(false);
+
+  const [CloseReferralExplanation, setCloseReferralExplanation] = useState('');
+
+  useEffect(() => {
+    if (mutation.isSuccess) {
+      setIsCloseReferralModalOpen(false);
+      setIsFormCleaned(false);
+      setCloseReferralExplanation('');
+    }
+  }, [mutation.status]);
+
+  return (
+    <ReactModal
+      ariaHideApp={!isTestEnv}
+      isOpen={isCloseReferralModalOpen}
+      onRequestClose={() => setIsCloseReferralModalOpen(false)}
+      style={{
+        content: {
+          maxWidth: '32rem',
+          padding: '0',
+          position: 'static',
+        },
+        overlay: {
+          alignItems: 'center',
+          backgroundColor: 'rgba(0,0,0,0.75)',
+          display: 'flex',
+          flexDirection: 'column',
+          justifyContent: 'center',
+        },
+      }}
+    >
+      <form
+        aria-labelledby={seed('close-referral-form')}
+        onSubmit={(e) => {
+          e.preventDefault();
+          setIsFormCleaned(true);
+          if (CloseReferralExplanation.length > 0) {
+            mutation.mutate({
+              action: 'close_referral',
+              payload: {
+                close_explanation: CloseReferralExplanation,
+              },
+              referral,
+            });
+          }
+        }}
+      >
+        <div className="p-8 space-y-4">
+          <h2 className="text-xl" id={seed('close-referral-form')}>
+            <FormattedMessage {...messages.modalTitle} />
+          </h2>
+
```

Comment:
IMO we need to add a small explanation text here.

"The closing of referrals is intended for invalid referrals or those that do not meet the criteria for obtaining an answer. Please help the requester by explaining to them why this referral is not valid."